snapshot_dbg_cli/delete_debuggees_command.py

```diff
@@ -0,0 +1,167 @@
+"""
+
+QUIET_HELP = 'If set, suppresses user confirmation of the command.'
+
+SUMMARY_HEADERS = ['Name', 'ID', 'Last Active', 'Status']
+
+
+def transform_to_debuggee_summary(debuggee):
+  # Match the fields from SUMMARY_HEADERS
+  return [
+      debuggee['displayName'],
+      debuggee['id'],
+      debuggee['lastUpdateTime'],
+      get_debuggee_status(debuggee),
+  ]
+
+
+def get_debuggee_status(debuggee):
+  if not debuggee['activeDebuggeeEnabled']:
+    return 'WARNING UNKNOWN'
+
+  if debuggee['isActive']:
+    return 'ACTIVE'
+
+  if not debuggee['isStale']:
+    return 'INACTIVE'
+
+  return 'STALE'
+
+
+def should_delete_debuggee_check(debuggee, args):
+  if args.include_all:
+    return True
+
+  if args.include_inactive:
+    return not debuggee['isActive']
+
+  # By default we only delete stale debuggees
+  return debuggee['isStale']
+
+
+class DeleteDebuggeesCommand:
+  """This class implements the delete_debuggees command.
+
+  The register() method is called by the CLI startup code to install the
+  delete_debuggees command information, and the cmd() function will be invoked
+  if the delete_debuggees command was specified by the user.
+  """
+
+  def __init__(self):
+    pass
+
+  def register(self, args_subparsers, required_parsers, common_parsers):
+    parent_parsers = [common_parsers.database_url, common_parsers.format]
+    parent_parsers += required_parsers
+    parser = args_subparsers.add_parser(
+        'delete_debuggees', description=DESCRIPTION, parents=parent_parsers)
+    parser.add_argument('ID', help=ID_HELP, nargs='*')
+    parser.add_argument(
+        '--include-inactive', help=INCLUDE_INACTIVE_HELP, action='store_true')
+    parser.add_argument(
+        '--include-all', help=INCLUDE_ALL_HELP, action='store_true')
+    parser.add_argument('--quiet', help=QUIET_HELP, action='store_true')
+    parser.set_defaults(func=self.cmd)
+
+  def cmd(self, args, cli_services):
+    user_input = cli_services.user_input
+    user_output = cli_services.user_output
+    debugger_rtdb_service = cli_services.get_snapshot_debugger_rtdb_service()
+
+    # This will be a list, if no IDs were specified it will be empty. If any IDs
+    # are specified those are the only ones that will be deleted.
+    debuggee_ids = args.ID
+
+    debuggees = []
+    current_time_unix_msec = get_current_time_unix_msec()
+
+    if debuggee_ids:
+      ids_not_found = []
+
+      for debuggee_id in debuggee_ids:
+        debuggee = debugger_rtdb_service.get_debuggee(debuggee_id,
+                                                      current_time_unix_msec)
+        if debuggee is None:
+          ids_not_found.append(debuggee_id)
+        else:
+          debuggees.append(debuggee)
+
+      if ids_not_found:
+        user_output.error('Debuggee ID not found: '
+                          f"{', '.join(ids_not_found)}")
+        raise SilentlyExitError
+    else:
+      debugger_rtdb_service = cli_services.get_snapshot_debugger_rtdb_service()
```

Comment:
This can be removed; it's in the common code above

Reply:
good catch